dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs

```diff
@@ -424,10 +444,25 @@ await foreach (Completions completions in response.ConfigureAwait(false))
                     s_inflightAutoInvokes.Value--;
                 }
 
-                var stringResult = ProcessFunctionResult(functionResult, chatExecutionSettings.ToolCallBehavior);
+                // Apply any changes from the function call filters context to final result.
+                functionResult = new FunctionResult(invocationContext.Result);
+
+                object functionResultValue = functionResult.GetValue<object>() ?? string.Empty;
+                var stringResult = ProcessFunctionResult(functionResultValue, chatExecutionSettings.ToolCallBehavior);
 
                 AddResponseMessage(chatOptions, chat, stringResult, errorMessage: null, toolCall.Id, this.Logger);
 
+                // If filter requested to stop function call iteration, breaking function calling loop.
+                if (invocationContext.Action.HasFlag(FunctionCallAction.StopFunctionCallIteration))
```

Comment:
The iterator variable is called `toolCallIteration` but the flag we are using is called `StopFunctionCallIteration`. Should we align our terminology?

Reply:
There are no more flags in latest version, so we should be good. Regarding iterators in context, I renamed them to `RequestSequenceNumber` and `FunctionSequenceNumber`, happy to discuss this naming if needed.